File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if (nums.length === 0) return 0
        const checked = new Map()

        for (const num of nums) {
            checked.set(num, false)
        }
        let result = 1;

        for (const num of nums) {
            let current = num
            let existed = checked.has(current) && checked.get(current) === false
            if (!existed) continue
            checked.set(current, true)
            let max = 1
            while (existed) {
                existed = checked.has(current + 1)
                if (existed) {
                    checked.set(current + 1, true)
                    current = current + 1
                    max++
                }
            }

        

            result = Math.max(max, result)
        }

        return result
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        dic = {}
        for num in nums:
            c = str(num)
            if c not in dic:
                dic[c] = 1
            else:
                dic[c] = dic[c] + 1
                return True
        
        return False

        